sklearn/metrics/_classification.py

```diff
@@ -2030,55 +2077,135 @@ class after being classified as negative. This is the case when the
             f"problems, got targets of type: {y_type}"
         )
 
+    msg_deprecated_param = (
+        "`raise_warning` was deprecated in version 1.7 and will be "
+        "removed in 1.9, when an `UndefinedMetricWarning` will always raise in "
+        "case of a division by zero and the value set with the "
+        "`replace_undefined_by` param will be returned."
```

Comment:
I would cut the sentence to decouple the deprecation warning from the future behaviour to make it easier to read.

```suggestion
        "`raise_warning` was deprecated in version 1.7 and will be "
        "removed in 1.9. An `UndefinedMetricWarning` will always be raised in "
        "case of a division by zero and the value set with the "
        "`replace_undefined_by` param will be returned."
```

Do you think that we should mentioned that the warning could be filtered with `warnings.filterwarnings` to have the equivalence with the previous `raise_warning`?